source/common/http/codec_client.cc

```diff
@@ -136,11 +136,12 @@ void CodecClient::onData(Buffer::Instance& data) {
 
 CodecClientProd::CodecClientProd(Type type, Network::ClientConnectionPtr&& connection,
                                  Upstream::HostDescriptionConstSharedPtr host,
-                                 Event::Dispatcher& dispatcher)
+                                 Event::Dispatcher& dispatcher, bool validate_header_values)
```

Comment:
maybe strict_header_validation?
since we were validating a bit before, just not much

Reply:
Hm, or maybe I'm pinging when you're not ready for me to take another pass.  Let me know when I should take a look?  :-)

Reply:
Sure, I think `strict_header_validation` is a slightly more accurate description of what's going on.  

`validate_header_values` is consistent with the runtime feature name, and all the other parameters/contructor args however.  Do you think I should find/replace them all, so we standardize on `strict_header_validation`?

Reply:
That'd be great, if you're up for it.

Reply:
Did a PR-wide `s/validate_header_values/strict_header_validation/`